src/io/rsm/raft.hpp

```diff
@@ -19,17 +19,23 @@
 #include <map>
 #include <set>
 #include <thread>
+#include <type_traits>
 #include <vector>
 
 #include <boost/core/demangle.hpp>
 
 #include "io/message_conversion.hpp"
+#include "io/rsm/shard_rsm.hpp"
```

Comment:
I think introducing a concrete RSM implementation as a dependency to the general Raft implementation is a bad idea because of the circular dependencies.